### PR TITLE
Update branding to 6.0.0 Alpha1

### PR DESCRIPTION
**Pull request by dougbu**
- hold the TFM back at `net5.0`
- update site extensions to split 5.0 and 6.0
- remove unnecessary `@(SuppressBaselineReference)` items
  - either 3.1.7 baselines we're using don't include reference or still using package
  - fix some comments and `Condition` attributes to make remainder easy to find
  - I'll undo any changes that were over-zealous

### CODE REVIEW

File: eng/Versions.props
Comment (dougbu) [inline eng/Versions.props:242]: Should this be 2.2.5 now? Or, can this package be removed here, in Dependencies.props and in Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj❔

File: eng/Versions.props
Comment (dougbu) [inline eng/Versions.props:243]: Should this be 3.1.7-servicing-20372-13 now❔ That shipped (sort of) with 3.1.7.

File: eng/Versions.props
Comment (dougbu) [inline eng/Versions.props:246]: /fyi this is the version that shipped (sort of) with 5.0.0-preview-7.

File: src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj
Comment (dougbu) [inline src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:60]: This and the `@(PackageReference)` items in release/5.0 need to use `5.0`. Is that on your team @BrennanConroy❔

File: eng/Versions.props
Comment (BrennanConroy) [reply eng/Versions.props:242]: I'm all for removing anything 2.2
Lets ping some folks to make sure they're ok with us removing 2.2 from the site extension
@Pilchie @shirhatti @bradygaster

File: src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj
Comment (BrennanConroy) [reply src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:60]: Did you mean to put this comment on the other side of the diff?

I'll take a look at 5.0 and update 3.1 -> 5.0 if needed

Comment (dougbu) [reply src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:60]: Much appreciated❕ Hopefully 3.1 isn't busted in this respect too.

File: eng/Versions.props
Comment (dougbu) [reply eng/Versions.props:243]: @BrennanConroy is this your call❔ If not you, who❔

File: src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj
Comment (BrennanConroy) [reply src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:60]: I don't think it is, that's the one that was fixed a couple months ago and I think the merge forward kept this as 3.1

File: eng/Versions.props
Comment (BrennanConroy) [reply eng/Versions.props:243]: Likely my call, I think it doesn't matter if it stays or updates (assuming the 3.1.3-servicing bits stay in the feed). I'll double check tomorrow.

**Comment by dougbu** [comment]
/fyi This PR needs more work to get the tests passing ☹️

File: src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj
Comment (BrennanConroy) [reply src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:60]: Technically 5.0 should probably be updated, but it's currently producing the correct package. I guess I'll update it for consistency.

File: eng/Versions.props
Comment (BrennanConroy) [reply eng/Versions.props:243]: We probably should keep this as the latest. Anyway to automate that?

Comment (dougbu) [reply eng/Versions.props:243]: Doubt this is possible to automate given oddities with circular subscriptions. But, let's chat w/ @mmitche and @wtgodbe in the 3.1.9 timeframe. Could you file an issue while I'm thinking of other problems with this PR please❔ (Placing it in https://github.com/orgs/dotnet/projects/10 will help us remember.)

File: eng/AfterSolutionBuild.targets
Comment (dougbu) [inline eng/AfterSolutionBuild.targets:18]: Use of `$(AspNetCoreMajorVersion)` or `$(AspNetCoreMajorMinorVersion)` was only occasionally correct. Since the concepts aren't related, we shouldn't change this back in the future

File: eng/targets/Helix.props
Comment (dougbu) [inline eng/targets/Helix.props:25]: Note to self: If we need another update, use `$(AspNetCoreMajorMinorVersion).$(AspNetCorePatchVersion)` here

File: src/Identity/Extensions.Stores/src/Microsoft.Extensions.Identity.Stores.csproj
Comment (dougbu) [inline src/Identity/Extensions.Stores/src/Microsoft.Extensions.Identity.Stores.csproj:16]: Note to self: If we need another iteration, fix indentation here

**Comment by dougbu** [comment]
Internal build https://dev.azure.com/dnceng/internal/_build/results?buildId=778919 should help confirm changes in site extensions build

**Comment by dougbu** [comment]
:up::date: though I doubt tests will work without further changes. I'm building locally to see if I can get more information about template failures in previous builds. Suspect we have additional places where TFMs get mixed up with package versions.

Worst case, we might need a runtime update containing dotnet/runtime#40950. I triggered that (daily) subscription just in case.

**Comment by dougbu** [comment]
@pranavkm I seem to remember you looking at test failures like and that you resolved them. 
``` text
The active test run was aborted. Reason: Test host process crashed : Unhandled exception. System.InvalidOperationException: There is no currently active test.
```
Any idea why I'm seeing them in the gRPC interop tests e.g. https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-24983-merge-c76e36077044499e8d/InteropTests--net5.0/console.74b37675.log?sv=2019-02-02&se=2020-09-09T20%3A09%3A24Z&sr=c&sp=rl&sig=89ThpfLWPzA57TSNhBUDUXyo5Zl7VM%2FE5dV4aEzYBKI%3D❔

**Comment by dougbu** [comment]
@HaoK I'm seeing gRPC test failures on other platforms that seem to indicate the newly-built 6.0.0-ci copy of Microsoft.AspNetCore.App.Runtime was installed to the wrong location. What needs to change❔

For example, from https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-24983-merge-e4e47cb7c8b8484ca9/InteropTests--net5.0/console.ab71631b.log?sv=2019-02-02&se=2020-09-09T20%3A08%3A14Z&sr=c&sp=rl&sig=YqU17c32Tp8rRlY3EZdxU9Vo%2BSHqmJYs5K7gpC6QeuE%3D:
``` text
Found AspNetRuntime: Microsoft.AspNetCore.App.Runtime.win-x64.6.0.0-ci.nupkg, extracting *.txt,json,dll to /datadisks/disk1/work/B8880A11/w/B18309D8/e/.dotnet28088/shared/Microsoft.AspNetCore.App/5.0.0-rc.1.20417.14
```

**Comment by HaoK** [comment]
The versions haven't been updated yet? `-sdk 5.0.100-rc.1.20379.10 --runtime 5.0.0-rc.1.20417.14`

https://github.com/dotnet/aspnetcore/blob/master/eng/targets/Helix.targets#L120  I think we use MicrosoftNETCoreAppRuntimeVersion

**Comment by dougbu** [comment]
> The versions haven't been updated yet? `-sdk 5.0.100-rc.1.20379.10 --runtime 5.0.0-rc.1.20417.14`

We should be able to rebrand independent of the dotnet/runtime repo. Helix.targets and the scripts used on the Helix agents should _not_ use `$(MicrosoftNETCoreAppRuntimeVersion)` when installing Microsoft.AspNetCore.App.Runtime. Guess I've got another thing to separate in this PR&hellip;

----

/fyi the same messed up install location for the new Microsoft.AspNetCore.App.Runtime bits is hurting the project template tests too e.g. https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-24983-merge-56d6b0427e38425e94/ProjectTemplates.Tests--net5.0/console.2800c965.log?sv=2019-02-02&se=2020-09-09T20%3A09%3A23Z&sr=c&sp=rl&sig=fJ7Q1BIJ1qJNSQxW8J2SP801r31rmo3kHfbakOY0szw%3D

On Windows, the project template downstream errors are usually "failed to run EF migrations" due to an inability to load the Microsoft.AspNetCore.Diagnostics.Abstractions.dll assembly that Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.dll needs. The gRPC downstream errors are thrown `TimeoutException`s.

**Comment by HaoK** [comment]
Right both GRPC and templates depend on the aspnet runtime, that currently use that version, i think if you just replace that with whatever the aspnet specific version is, things shoudl work

**Comment by dougbu** [comment]
@HaoK please have a look @ the last commit

File: eng/targets/Helix.props
Comment (dougbu) [reply eng/targets/Helix.props]: Property removed instead

File: eng/targets/Helix.targets
Comment (dougbu) [inline eng/targets/Helix.targets:20]: What needs every package we ship❔ And, does including these packages in the correlation payload mean they're available when restoring packages❔ This isn't reliable in any case because it's often evaluated before projects build or publish.

**Comment by dougbu** [comment]
> @HaoK please have a look @ the last commit

Now the second-to-last commit

File: eng/targets/Helix.targets
Comment (HaoK) [reply eng/targets/Helix.targets:20]: The template tests use a wide variety of packages we build, rather than listing them all, it seems reasonable for us to just install the ones we build, these aren't in the correlation payload, they show up in the workitem root

**Comment by dougbu** [comment]
New test internal build is https://dev.azure.com/dnceng/internal/_build/results?buildId=781173 Previous one didn't uncover any issues in the packages but I'd like to double-check

**Comment by HaoK** [comment]
If you say SharedFxVersion is the right version to use, I'm guessing it is :)  Looks fine to me

File: eng/targets/Helix.targets
Comment (dougbu) [reply eng/targets/Helix.targets:20]: Ah,
``` text
nuget add source {Options.HELIX_WORKITEM_ROOT} --configfile NuGet.config 
```

**Comment by dougbu** [comment]
Thanx @HaoK❕ 

@JunTaoLuo or @wtgodbe please review this PR before we get too far into tomorrow. And, code owners please chime in if something in here skews your world.

**Comment by dougbu** [comment]
`dotnet ef` commands in project template tests are failing with similar `FileLoadException`s to the Blazor WASM tests. Project template exceptions are about Microsoft.AspNetCore.Diagnostics.Abstractions.dll. Unfortunately these problems did not result in `*.dmp` files.

gRPC interop tests now failing with `BadImageFormatException`s. `*.dmp` files may help here.

File: src/ProjectTemplates/Web.ProjectTemplates/Microsoft.DotNet.Web.ProjectTemplates.csproj
Comment (cartermp) [reply src/ProjectTemplates/Web.ProjectTemplates/Microsoft.DotNet.Web.ProjectTemplates.csproj]: Actually two problems:

1. This not being in the template means that the worker service does not build
2. It should be a .fsproj, like in the PR here: https://github.com/dotnet/aspnetcore/pull/22042/files#diff-77ac0b0b902172169ed9b7307e56cf52 - so something went awry

Comment (dougbu) [reply src/ProjectTemplates/Web.ProjectTemplates/Microsoft.DotNet.Web.ProjectTemplates.csproj]: Given problems getting this PR working, I'm thinking about separating this one-line fix into another PR. For the first point, the `WorkerTemplateAsync()` test checks only C#. I'll try adding another test if I create a new PR tonight or (more likely) tomorrow. (Want to see if I can find the cause of the image loading issues w/ Fusion logging first.)

I fixed the second problem.

**Comment by dougbu** [comment]
FYI @captainsafia @javiercn @pranavkm I'm having no luck getting Fusion logs for the Blazor WASM `FileLoadException` and haven't downloaded a `*.dmp` from the most recent failures on Helix.

File: src/ProjectTemplates/Web.ProjectTemplates/Microsoft.DotNet.Web.ProjectTemplates.csproj
Comment (dougbu) [reply src/ProjectTemplates/Web.ProjectTemplates/Microsoft.DotNet.Web.ProjectTemplates.csproj]: Moved this change to #25180.

**Comment by dougbu** [comment]
:up::date: to remove the three commits I moved to #25180 and #25181. None of those commits is _required_ in this PR. They were here only because I found the problems while working on the branding and the branding / TFM confusion.

**Comment by dougbu** [comment]
/fyi @mmitche this probably needs another iteration but is much closer than it was.

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj
Comment (dougbu) [inline src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:63]: @rainersigwald I added `AfterTargets` because the `ResolveFrameworkReferences` has a `Returns` and thought that indicated it might get used from an outer build. But, that doesn't seem to be the case, at least in the cases I checked. In addition, this project has just a single target framework as seen above -- https://github.com/dotnet/aspnetcore/pull/24983/files#diff-eef634c0ab4b1d011d44f35afc0ed324R4
So, should we need both `AfterTargets` and `DependsOnTargets`❔

FYI the overall aim here is to make sure we fix the resolved targeting pack and framework reference metadata before anything tries to use it.

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj
Comment (dougbu) [inline src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:21]: @pranavkm this project is a mess and I don't remember why it needs these hacks. the .nuspec will contain just the framework reference no matter what we do (due to the `RemoveSharedFrameworkOnlyRefsFromNuspec` fixes in eng/AfterSolutionBuild.targets) and anything that lands in the .deps.json file (or whatever) will be overridden using the targeting pack's PackageOverrides.txt file in user's projects i.e. packages won't be downloaded.

Please remind me: Why do we need this unique special case❔

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj
Comment (rainersigwald) [reply src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:63]: Having it listed in both is not unreasonable.

`AfterTargets` means "whenever `ResolveFrameworkReferences` runs, run this right afterward."

`DependsOnTargets` means "if this is going to run, run `ResolveFrameworkReferences` first."

In the normal flow of things, `ResolveFrameworkReferences` runs because of other dependencies (in the common targets/the SDK). So your `AfterTargets` probably generally hooks to that, and might be sufficient on its own.

`DependsOnTargets` would make sense if the target dependency flow goes through this target--because one of its `BeforeTargets` wants to run, so this runs, and the `DependsOn` forces `ResolveFrameworkReferences` to run.

**Comment by dougbu** [comment]
/ping reviewers

This PR addresses additional version confusions that have crept into the repo, specifically mixing the default .NET TFM with the shared framework version. Switching to 6.0.0 does not mean we're targeting net6.0 (yet).

This PR also introduces follow-up items #26448 and #26776.

**Comment by dougbu** [comment]
PR is green.

Only aspnetcore-quarantined-pr tests failed in last validation runs. Failures in https://dev.azure.com/dnceng/public/_build/results?buildId=848137 look very similar to those in https://dev.azure.com/dnceng/public/_build/results?buildId=848129 (the last rolling build of the master branch), including the Helix Linux and Windows timeouts. IOW there's no new failures in this PR compared to the existing master branch.

File: src/Framework/Directory.Build.props
Comment (JunTaoLuo) [reply src/Framework/Directory.Build.props:18]: This doesn't look resilient. Granted we probably won't change the format netX.Y to something else, I guess I'm okay with it.

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj
Comment (JunTaoLuo) [inline src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:25]: Why did this reference change? I'm not too sure about the changes in this file but as long as @pranavkm signs off it should be fine.

**Review by JunTaoLuo — APPROVED** [review]

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj
Comment (mmitche) [reply src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:25]: Probably putting an error above that indicating that we expect "netN" so that we catch this location if something changes.

Comment (dougbu) [reply src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:25]: The whole project changed because we have a mess of conflicts because this project attempted to use AspNetCore bits from the SDK as well as reference projects elsewhere in the repo. @pranavkm signed off on doing this in Teams.

 #26448 is about trimming the package _correctly_ at some point in the 6.0 train but only if this project doesn't move elsewhere.

**Pull request merged**
